Define a security policy for geotools project (#4187)
geotools/geotools: SECURITY.md

# Security Policy

## Supported Versions

| Version | Supported          | Policy |
| ------- | ------------------ | ------- |
| 28.x   | ✅ | Supported branch (6 months from initial release) |
| 27.x   | ⚠️                | Maintenance branch (6 additional months) |
| 26.x   | :x:               | Unsupported (volunteers release as needed) |

## Reporting a Vulnerability

As a Java library we often find vulnerabilities are reported by downstream (and thus user facing) projects.

> * We ask that vulnerabilities be reported via our bi-weekly video chat meeting (see developer list for invite).

Author: ```suggestion
* We ask that vulnerabilities be reported via our bi-weekly video chat meeting (see [developer list](https://geotools.org/getinvolved.html) for meeting invite).
```

> * We ask that vulnerabilities be reported via our bi-weekly video chat meeting (see [developer list](https://geotools.org/getinvolved.html) for meeting invite).

Author: ```suggestion
* GeoTools uses GitHub [security](https://github.com/geotools/geotools/security) option for *private vulnerability reporting*
* To discuss vulnerabilities attend our bi-weekly video chat meeting (see [developer list](https://geotools.org/getinvolved.html) for meeting invite).
```

# Security Policy

## Supported Versions

| Version | Supported          | Policy |
| ------- | ------------------ | ------- |
| 28.x   | ✅ | Supported branch (6 months from initial release) |
| 27.x   | ⚠️                | Maintenance branch (6 additional months) |
| 26.x   | :x:               | Unsupported (volunteers release as needed) |

## Reporting a Vulnerability

> As a Java library we often find vulernabilities are reported by downstream (and thus user facing) projects.

Reviewer: ```suggestion
As a Java library we often find vulnerabilities are reported by downstream (and thus user facing) projects.
```

> * We ask that vulernabilities be reported via our bi-weekly video chat meeting (see developer list for invite).

Reviewer: ```suggestion
* We ask that vulnerabilities be reported via our bi-weekly video chat meeting (see developer list for invite).
```
A link to the developer list/archive should be provided

* GeoTools is an Open Source Geospatial Foundation project, the [geotools project officer](https://www.osgeo.org/about/board/) can be emailed directly if you are unable to attend meeting above.

> * New: We have enabled GitHub Security Advisories allowing secure reporting of vulnerabilities directly to the projec team

Author: It does not look like security vulnerabilities can be opened to report issues.
```suggestion
* GitHub Security Advisories list reported / patched vulnerabilities
```

Reviewer: that's not how it works; a GH project admin can create a security advisory: https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/about-repository-security-advisories

It would be good  to enable private reporting: https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability - there should be a button "Private vulnerability reporting" on https://github.com/geotools/geotools/settings/security_analysis

Author: How do we set up a private reporting option? Is it just with GitHub issues?

> To allow downstream projects an oppertunity to upgrade the library issue tracker uses placeholder issues (with no description or details). Details are added when fix is available in supported and maintenance branches above.

Reviewer: ```suggestion
To allow downstream projects an opportunity to upgrade the GeoTools issue tracker uses placeholder issues (with no description or details). Details are added when fix is available in the supported and maintenance branches listed above.
```
